iceoryx_utils/include/iceoryx_utils/internal/cxx/unique_ptr.inl

```diff
@@ -0,0 +1,164 @@
+
+#ifndef IOX_UTILS_CXX_UNIQUE_PTR_INL
+#define IOX_UTILS_CXX_UNIQUE_PTR_INL
+
+#include "iceoryx_utils/cxx/unique_ptr.hpp"
+
+namespace iox
+{
+namespace cxx
+{
+
+template<typename T>
+unique_ptr<T>::unique_ptr(std::function<void(T*)>&& deleter) noexcept : m_deleter(deleter)
+{}
+
+template<typename T>
+unique_ptr<T>::unique_ptr(ptr_t ptr, std::function<void(T*)>&& deleter) noexcept : m_ptr(ptr), m_deleter(std::move(deleter))
+{}
+
+// Can't have this function because the signature clashes with the case where T = void.
+// If we definitely want it, need to make it considered only when T is not void...
+//template<typename T>
+//unique_ptr<T>::unique_ptr(void* allocation, std::function<void(T*)>&& deleter) noexcept
+//    : m_ptr(reinterpret_cast<T*>(allocation)), m_deleter(deleter)
+//{}
```

Comment:
Remove and disallow use of `void`?

Reply:
I would like to address this in a separate PR when I flesh out the unique_ptr implementation.
I'll delete the commented code for now.